airflow/jobs/local_task_job.py

```diff
@@ -154,11 +148,13 @@ def handle_task_exit(self, return_code: int) -> None:
         # in case it failed due to runtime exception/error
```

Comment:
This comment ("task exited by itself" part) is no longer true as it is called from the signal handler